packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadView/useUploadView.ts

```diff
@@ -3,30 +3,57 @@ import React from 'react';
 import { uploadHandler } from '../../../actions';
 
 import { useGetActionInput } from '../../../providers/configuration';
-import { useStore } from '../../../providers/store';
+import { FileItems, useStore } from '../../../providers/store';
 import { Task, useProcessTasks } from '../../../tasks';
 
 import { DEFAULT_ACTION_CONCURRENCY } from '../constants';
 import { UploadViewState, UseUploadViewOptions } from './types';
 import { DEFAULT_OVERWRITE_ENABLED } from './constants';
+import { isUndefined } from '@aws-amplify/ui';
+import { isFileTooBig } from '../../../validators';
 
 export const useUploadView = (
   options?: UseUploadViewOptions
 ): UploadViewState => {
+  const [invalidFiles, setInvalidFiles] = React.useState<
+    FileItems | undefined
+  >();
   const { onExit: _onExit } = options ?? {};
   const getInput = useGetActionInput();
   const [{ files, location }, dispatchStoreAction] = useStore();
   const { current, key } = location;
 
+  const validFiles = React.useMemo(() => {
+    const validFileItems: FileItems = [];
+    files?.forEach((fileItem) => {
+      const { id, file } = fileItem;
+      if (isFileTooBig(file)) {
+        setInvalidFiles((prev) =>
+          isUndefined(prev) ? [fileItem] : prev.concat(fileItem)
+        );
+        dispatchStoreAction({ type: 'REMOVE_FILE_ITEM', id });
+      } else {
+        validFileItems.push(fileItem);
+      }
+    });
+
+    return validFileItems;
+  }, [files, dispatchStoreAction]);
```

Comment:
We can remove the side effect behaviors by returning both valid files and invalid files here:
```
const { validFiles, invalidFiles } = React.useMemo(() => ..);
```

We can clean out the invalid files from the store by exposing an `onMessageDismiss` callback on the return of `useUploadView`, something like:
```
const onMessageDismiss = () => {
  invalidFiles?.map((: data }) =>  dispatchStoreAction({ type: 'REMOVE_FILE_ITEM', id: data.id });
}
```

And provide it to the controls context when `invalidFiles` is truthy

Reply:
However this will make the invalid files exist in the files context longer than we want. Currently they are removed when the upload started. Now we want it to be removed only when the message is dismissed?